### PR TITLE
Expose allow broken links validation setting

### PR DESCRIPTION
**Pull request by kdorosh**
# Description

Some users have noted that with validation enabled, resources that reference non-existent resources are accepted. (and are marked with a warning status)

This change exposes a toggle that allows validation to reject resources on warning, not just on errors.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3099

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick https://console.cloud.google.com/cloud-build/builds/24c44791-b47e-4323-bc5a-dd5db494c998;step=14?project=solo-public regression test flake

File: projects/gateway/pkg/validation/validator_test.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ var _ = Describe("Validator", func() {
 					Expect(proxyReports).To(HaveLen(0))
 				})
 			})
-			Context("allowBrokenLinks=true", func() {
+			Context("allowWarnings=true", func() {
```
Comment (arianaw66) [inline projects/gateway/pkg/validation/validator_test.go:107]: Do we have a test for the `false` case?

File: projects/gloo/api/v1/settings.proto [hunk 2]
```diff
@@ -419,6 +419,11 @@ message GatewayOptions {
         // Validation will still log the error and increment the validation.gateway.solo.io/resources_rejected stat
         // Currently defaults to true - must be set to `false` to prevent writing invalid resources to storage
         google.protobuf.BoolValue always_accept = 6;
+
+        // Accept resources if validation produced a warning (defaults to true)
```
Comment (arianaw66) [inline projects/gloo/api/v1/settings.proto:423]: This way it'll be more readable in the proto in the docs.
```suggestion
        // Accept resources if validation produced a warning (defaults to true).
```

File: projects/gloo/api/v1/settings.proto [hunk 3]
```diff
@@ -419,6 +419,11 @@ message GatewayOptions {
         // Validation will still log the error and increment the validation.gateway.solo.io/resources_rejected stat
```
Comment (arianaw66) [inline projects/gloo/api/v1/settings.proto:419]: And let's fix this one and the one on the previous line while we're at it.
```suggestion
        // Validation will still log the error and increment the validation.gateway.solo.io/resources_rejected stat.
```

File: projects/gateway/pkg/validation/validator_test.go (hunk 1 above)
Comment (kdorosh) [reply projects/gateway/pkg/validation/validator_test.go:107]: the rest of the test suite actually 😄 so several tests

this context overrides the validator with the setting change, but by default our tests are running with `allowWarnings`=`false`

Comment (arianaw66) [reply projects/gateway/pkg/validation/validator_test.go:107]: Oh, great :)

**Comment by kdorosh** [comment]
/kick regression test flake, not consistent? https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=e84eeddb-b862-4684-a307-ba7a1e31c1be

**Pull request merged**